### PR TITLE
Fixing - IDENTITY-4154, IDENTITY-4152

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
https://wso2.org/jira/browse/IDENTITY-4154
https://wso2.org/jira/browse/IDENTITY-4152

Added the facility to show error messages without page refresh.
Moved the user exist search to a service.
Fixed the footer misalignment scenarios. [(IDENTITY-4152)] (https://wso2.org/jira/browse/IDENTITY-4152)
Added a response to show to user after successful signup.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/registration.jsp [hunk 1]
```diff
@@ -14,54 +14,87 @@
...
-    int errorCode = 0;
     try {
+
         UserRegistrationAdminServiceClient registrationClient = new UserRegistrationAdminServiceClient();
-        boolean isExistingUser = CarbonContext.getThreadLocalCarbonContext().getUserRealm().
-                getUserStoreManager().isExistingUser(request.getParameter("reg-username"));
-        if (isExistingUser){
-            errorCode = 1;
-            throw new Exception();
+        boolean isExistingUser = registrationClient.isUserExist(request.getParameter("reg_username"));
```
Comment (lalaji) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/registration.jsp]: We can assign request.getParameter("reg_username") to a variable and re-use.I noticed those parameter values  have re-use in different places

File: components/user-mgt/org.wso2.carbon.identity.user.registration/src/main/java/org/wso2/carbon/identity/user/registration/UserRegistrationService.java [hunk 2]
```diff
@@ -193,14 +194,32 @@ public boolean isAddUserWithInfoCardEnabled() throws Exception {
...
+     * @return True if exist.
+     * @throws Exception
+     */
+    public boolean isUserExist(String username) throws Exception {
+
+        try {
+            return CarbonContext.getThreadLocalCarbonContext().getUserRealm().
+                    getUserStoreManager().isExistingUser(username);
+        } catch (UserStoreException e) {
+            throw new Exception("User registration exception", e);
```
Comment (lalaji) [inline components/user-mgt/org.wso2.carbon.identity.user.registration/src/main/java/org/wso2/carbon/identity/user/registration/UserRegistrationService.java]: Better to improve error message

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/registration.jsp (hunk 1 above)
Comment (JKAUSHALYA) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/registration.jsp]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/user-mgt/org.wso2.carbon.identity.user.registration/src/main/java/org/wso2/carbon/identity/user/registration/UserRegistrationService.java (hunk 2 above)
Comment (JKAUSHALYA) [inline components/user-mgt/org.wso2.carbon.identity.user.registration/src/main/java/org/wso2/carbon/identity/user/registration/UserRegistrationService.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/pom.xml [hunk 3]
```diff
@@ -42,6 +42,12 @@
         </dependency>
         <dependency>
             <groupId>javax.servlet</groupId>
+            <artifactId>jsp-api</artifactId>
+            <version>2.0</version>
```
Comment (msmshariq) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/pom.xml]: Define dependency version in the parent pom and inherit those, if not it's a maintenance nightmare.

Comment (JKAUSHALYA) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/pom.xml]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

**Pull request merged**